test/cmdlineTests/debug_info_in_yul_and_evm_asm_print_all/output

```diff
@@ -194,7 +194,7 @@ object "C_6" {
                     {
                         if callvalue() { revert(_1, _1) }
                         if slt(add(calldatasize(), not(3)), _1) { revert(_1, _1) }
-                        return(memoryguard(0x80), _1)
+                        return(_1, _1)
```

Comment:
Although: how does the ``memoryguard`` vanish completely here? Didn't we make it non-removable?
Maybe we didn't, arguing exactly that it will always still be used somewhere in the end...
Hm... that's a problem - it vanishing entirely may prevent stack-to-memory...

Reply:
Yeah, IIRC, the idea was that if we make ``memoryguard`` movable, not non-removable, we may accumulate a lot of weird ``pop(memoryguard(...))``s over time... and at the time, the value actually being written to memory guaranteed that at least one would remain - but that's already no longer true since after the unused store eliminator...

One way to fix this would probably be a special kind of side-effect "can not be *entirely* removed"... or just generally a special case in the ``UnusedPruner``, i.e. have it keep track of how many ``memoryguard``s there are and only consider one removable, if it is not the last one, while otherwise keeping a ``pop(memoryguard(...))`` around...

But none of that is really nice :-)...

Reply:
Hm... one could argue:
If code contains a ``memoryguard`` before optimization, then if all ``memoryguard``s to vanish during optimization, that has to mean that all dynamic memory uses could be eliminated... and we can just add the original ``memoryguard`` back... but that's also a bit weird...